### ISSUE TITLE
Pattern exceed max length of 32

### ISSUE THREAD

**Issue by pigranada** [loilo/Fuse#68]
The pattern exceed max lenght of 32 is showing for the search query that has more than 32 characters with some unicode characters and in UTF-8 encoding. This most likely happen in `BitapSearch::__construct()` during the chunking of characters. Upon further checking, it seems that UTF-8 encoding has problems with PHP's substr and it might be better to use mb_substr, see [[ stackoverflow discussion | https://stackoverflow.com/questions/12446877/php-is-it-possible-to-correctly-substr-a-utf-8-string

```
$query = 'send more money to my pàypàl account';
$fuzzyMatcher = new Fuse(['Sample collection only, please ignore'], []);

$fuzzyMatcher->search($query);
```

**Comment by pigranada** [comment]
Yes that is the error that I'm getting when I run the code. Sorry I have missed something in the code. Here is a better sample where the "Pattern length exceeds max of 32" exception.

```
$fuzzyMatcher = new \Fuse\Fuse(
    [], // Initial keywords list
    [
        'findAllMatches' => true,
        'ignoreLocation' => true,
        'includeMatches' => true,
        'includeScore' => true,
        'threshold' => 0,
    ]
);

$list = ['This is a very long message that needs to be matched to a pattern. '];
$fuzzyMatcher->setCollection($list);

try {
    $query = mb_convert_encoding("chárêctërs chárêctërs chárêctërs chárêctërs chárêctërs chárêctërs chárêctërs", "UTF-8"); // Simulate strings encoded in UTF-8
    $fuzzyMatcher->search($query);
} catch (Exception $e) {
    echo($e->getMessage());
}
```

**Comment by pigranada** [comment]
I also tried converting UTF-8 encoded strings to ASCII and it does not show the error anymore. However, it will be causing some false negatives and false positives since the unicode characters was converted to `?` characters once converted. So a pure unicode character string for query and pattern will most likely be an exact match (e.g ááááá to 赡赡赡赡).

**Comment by loilo** [comment]
Yes, I can reproduce this. And the culprit actually appears to be the `subtr` call you mentioned. (Funny enough, I have used `mb_substr` all throughout the codebase, just missed that single spot. Annoying.)
I'll publish a fix as soon as possible. 👍

**Comment by loilo** [comment]
Alright, a fix for this has been published in v6.5.5. Thanks again for the report! 🙂
